.circleci/config.yml

```
@@ -9,6 +9,35 @@ version: 2.1
 #
 
 commands:
+  check-skip:
+    steps:
+      - run:
+          name: Check-skip
+          command: |
+            git log --max-count=1 --pretty=format:"%B" | tr "\n" " " > commit_message.txt
+            export COMMIT_MESSAGE=$(cat commit_message.txt);
+            echo "Got commit message:"
+            echo "${COMMIT_MESSAGE}"
+            if [[ -v CIRCLE_PULL_REQUEST ]] && ([[ "$COMMIT_MESSAGE" == *"[skip circle]"* ]] || [[ "$COMMIT_MESSAGE" == *"[circle skip]"* ]]); then
```

It doesn't appear that you have any use for `commit_message.txt` elsewhere (also renamed the var to match Azure):
```suggestion
            export git_log=$(git log --max-count=1 --pretty=format:"%B" | tr "\n" " ");
            echo "Got commit message:"
            echo "${git_log}"
            if [[ -v CIRCLE_PULL_REQUEST ]] && ([[ "$git_log" == *"[skip circle]"* ]] || [[ "$git_log" == *"[circle skip]"* ]]); then
```

---

This is how I had it originally (see https://github.com/matplotlib/matplotlib/pull/20680/commits/34890a05b0a44df93b9cc67045899f1a1aadaa08#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R32) but something about the pipe and variable setting did not work, so I went the easy way and just outputted the result to a file. My BASH skills failed me and I figured this was good enough. I can spend more time on it if you think it's worth it

---

But that worked? There was no skip circle in that commit:
https://app.circleci.com/pipelines/github/matplotlib/matplotlib/10538/workflows/2803f328-d95f-424d-8335-1fe0abececc3/jobs/62399/parallel-runs/0/steps/0-102

---

Ah, I see, the message is not what was in the commit. But I don't think that's a BASH problem. We must have checked out something funny? I'll look a little closer.

---

I think the problem is the `--skip=1`, which you're doing _before_ merging, so that gets you not the latest commit. The code now doesn't have a skip.

---

Hmm yes there must be something wrong because locally the line you wrote worked. I can also try in a few minutes. If you beat me to a fix, feel free to push!